app/controllers/api/v0/container_images_controller.rb

```diff
@@ -4,6 +4,10 @@ class ContainerImagesController < ApplicationController
       include Api::Mixins::IndexMixin
       include Api::Mixins::ShowMixin
 
+      def index
+        render json: model.where(list_params).includes(:container_image_tags => :tag).references(:container_image_tags => :tag)
```

Comment:
this will make sure we do only 1 query, then the model touching these takes it from the cache

Reply:
@bdunne it should not be super hard to generate this automatically upon some condition, like model.taggable?